WordPress/gutenberg
# Fix the update of link and suggestions in LinkControl

When switching between two anchors in the same text block, the text input from the edit link window will not show the correct value.
See video with bug here: https://www.loom.com/share/d18dba4f4c0c43b0adb33f943e77ce0f





## Description

When switching between two links, the LinkControl component does not unmount.
The value prop will change but the component does not react to this change and does
not show the updated link & suggestions.

This PR does 2 things:
- it accounts for the value change inside of LinkControl and updates it's internal state
- it modifies the update of the suggestions when the value changes. The component only handled a new empty value, showing the initial suggestions if the user was to remove the value inside the input. But because the value can change and not be empty, we need to update the suggestions for this new non-empty value.


## How has this been tested?



- go to a page and add two links in the same text block
- go to the first link and click it to see the edit link window
- click the Edit button and see the text input with the URL and the suggestions list
- without clicking anything else, click the second link
- observe the values in the text input and the suggestion

## Screenshots 

## Types of changes




Bug fix

## Checklist:
